### PR TITLE
Fix SamplerResult num_clbits

### PR DESCRIPTION
**Pull request by ikkoham**
### Summary

Takeover #1668

### CODE REVIEW

File: test/terra/primitives/test_sampler.py [hunk 1]
```diff
@@ -525,6 +525,22 @@ def test_with_shots_none(self):
...
 
+    @data(1024, None)
+    def test_num_clbits(self, shots):
+        """test of QuasiDistribution"""
+        qc = QuantumCircuit(4)
+        qc.h(0)
+        qc.measure_all()
+
+        result = (
+            Sampler(backend_options={"method": "statevector"})
```
Comment (ikkoham) [inline test/terra/primitives/test_sampler.py:536]:
```suggestion
            Sampler()
```

**Review by t-imamichi — CHANGES_REQUESTED** [review]
LGTM

**Review by hhorii — COMMENTED** [review]
Though this may not be directly related to  this issue, maybe there is no error handling in Sampler. I'm wondering test cases can be added in this PR.

File: qiskit_aer/primitives/sampler.py [hunk 2]
```diff
@@ -125,10 +125,14 @@ def _call(
         for i in range(len(experiments)):
             if is_shots_none:
                 probabilities = result.data(i)["probabilities"]
-                quasis.append(QuasiDistribution(probabilities))
+                num_qubits = result.results[i].metadata["num_qubits"]
```
[the remark this answers is not in the document]
Comment (hhorii) [reply qiskit_aer/primitives/sampler.py:128]: Sorry. Once simulation is failed, `result.data(i)` raises an exception. In other words, Sampler will not return a partially-successful result (that includes both of success and fail). Is it ok?

[the remark this answers is not in the document]
Comment (hhorii) [reply qiskit_aer/primitives/sampler.py:128]: I talked with @ikkoham and he mentioned that `Sampler` does not support partially-successful result.

**Review by hhorii — APPROVED** [review]

**Pull request merged**